CodingTracker.Furiax/CodingTracker.Furiax/Crud.cs

```diff
@@ -0,0 +1,263 @@
+		{
+			TimeSpan timeBetween = CalculateDuration(startTime, stopTime);
+			string duration = timeBetween.ToString(@"hh\:mm");
+			if (timeBetween.Minutes <= 0)
+			{
+                Console.WriteLine("Session was less then a minute and therefore not stored");
+            }
+			else
+			{
+				using (var connection = new SqliteConnection(connectionString))
+				{
+					connection.Open();
+					var command = connection.CreateCommand();
+					command.CommandText = $"INSERT INTO CodeTracker (StartTime, EndTime, Duration) VALUES ('{start}', '{stop}', '{duration}')";
+					command.ExecuteNonQuery();
+					connection.Close();
+				}
+				Console.WriteLine("Record added to database");
+			}
+        }
+		internal static void CreateReport(string connectionString)
+		{
+			Console.Clear();
+			string sqlCommand = "SELECT * FROM CodeTracker";
+			List<CodingSession> sessions = new List<CodingSession>();
+			sessions = BuildList(connectionString, sqlCommand);
+			
+			TimeSpan totalTime = TimeSpan.Zero;
+			foreach (var session in sessions)
+			{
+					totalTime = totalTime + session.Duration;
+			}
+			TimeSpan averageTime = TimeSpan.FromTicks(totalTime.Ticks/sessions.Count);
+
+				Console.WriteLine($"You spend a total of {totalTime:%d} days, {totalTime:%h} hours and {totalTime:%m} minutes on coding, divided over {sessions.Count} sessions.");
+                Console.WriteLine($"On average you spend {averageTime:%h} hours, {averageTime:%m} minutes per session.");
+		}
+		internal static void GoalStatus(string connectionString, TimeSpan goalTime)
+		{
+			Console.Clear();
+			using (var connection = new SqliteConnection(connectionString))
+			{
+				var monday = DateTime.Today.AddDays(-(int)DateTime.Today.DayOfWeek + (int)DayOfWeek.Monday);
+				connection.Open();
+				var command = connection.CreateCommand();
+				command.CommandText = $"SELECT * FROM CodeTracker WHERE StartTime >= '{monday}'";
+				List<CodingSession> sessions = new List<CodingSession>();
+				SqliteDataReader reader = command.ExecuteReader();
+				while (reader.Read())
+				{
+					sessions.Add(new CodingSession
+					{
+						Id = reader.GetInt32(0),
+						StartTime = DateTime.ParseExact(reader.GetString(1), "dd/MM/yy HH:mm", new CultureInfo("nl-BE")),
+						EndTime = DateTime.ParseExact(reader.GetString(2), "dd/MM/yy HH:mm", new CultureInfo("nl-BE")),
+						Duration = TimeSpan.ParseExact(reader.GetString(3), @"hh\:mm", new CultureInfo("nl-BE"))
+					});
+				}
+				TimeSpan totalTimeCodedThisWeek = TimeSpan.Zero;
+				foreach (var session in sessions)
+				{
+					totalTimeCodedThisWeek += session.Duration;
+				}
+				int daysLeft = 7 - (int)DateTime.Today.DayOfWeek + 1;
+				if (goalTime <= totalTimeCodedThisWeek)
+				{
+					Console.WriteLine($"Goal achieved, you coded {totalTimeCodedThisWeek:%h} hours this week, while the goal was {goalTime.TotalHours} hours");
+				}
+				else
+				{
+					Console.WriteLine($"Progress: {totalTimeCodedThisWeek:%h}/{goalTime:%h} hours");
```

Comment:
Oops seems like I didn't do enough testing on that method (was the last one I wrote) 
I now completly reworked it cause the bad display above 24 wasn't the only thing that was faulty.
There were also older dates than the current week in the results, so that didn't work properly either.
I think I fixed all of this now, so I'm ready for a new review.